plugins/salesforce/core/src/main/java/org/pentaho/di/trans/steps/salesforce/SalesforceStep.java

```diff
@@ -142,4 +145,55 @@ public Object normalizeValue( ValueMetaInterface valueMeta, Object value ) throw
     }
     return value;
   }
+
+  @Override
+  public JSONObject doAction( String fieldName, StepMetaInterface stepMetaInterface, TransMeta transMeta,
+                              Trans trans ) {
+    JSONObject response = new JSONObject();
+    try {
+      Method actionMethod = SalesforceStep.class.getDeclaredMethod( fieldName + "Action" );
+      response = (JSONObject) actionMethod.invoke( this );
+      response.put( StepInterface.ACTION_STATUS, StepInterface.SUCCESS_RESPONSE );
+    } catch ( NoSuchMethodException | InvocationTargetException | IllegalAccessException e ) {
+      log.logError( e.getMessage() );
+      response.put( StepInterface.ACTION_STATUS, StepInterface.FAILURE_METHOD_NOT_RESPONSE );
+    }
+    return response;
+  }
+
+  @SuppressWarnings( "java:S1144" ) // Using reflection this method is being invoked
+  private JSONObject testButtonAction( ) {
+    JSONObject response = new JSONObject();
+    boolean successConnection = testConnection();
+    response.put( "connectionStatus", successConnection );
+    return response;
+  }
+
+  public boolean testConnection() {
+    boolean successConnection = true;
+    SalesforceConnection connection = null;
+
+    try {
+      String realURL = getTransMeta().environmentSubstitute( meta.getTargetURL() );
+      String realUsername = getTransMeta().environmentSubstitute( meta.getUsername() );
+      String realPassword = Utils.resolvePassword( getTransMeta(), meta.getPassword() );
+      int realTimeOut = Const.toInt( getTransMeta().environmentSubstitute( meta.getTimeout() ), 0 );
+
+      connection = new SalesforceConnection( log, realURL, realUsername, realPassword );
+      connection.setTimeOut( realTimeOut );
+      connection.connect();
+
+    } catch ( Exception e ) {
+      successConnection = false;
```

Comment:
we are not throwing the exception here so in the dialog class the is no point of catching and writing some code inside catch block

Reply:
We are not catching this exception in dialog. The try catch there is for a creating an instance of Meta class and it's needed.